src/hooks/useENSRegistrationActionHandler.ts

```diff
@@ -63,8 +68,17 @@ export default function useENSRegistrationActionHandler(
       : -1
   );
 
+  const avatarMetadata = useRecoilValue(avatarMetadataAtom);
+  const coverMetadata = useRecoilValue(coverMetadataAtom);
   const commitAction = useCallback(
     async (callback: () => void) => {
+      const changedRecords = await uploadRecordImages(
```

Comment:
looking good but can we move this upload to the register action? there could be cases where users would send the first tx (the commit tx) but won't continue with the flow so the name won't be registered